tools/edid/generate-rpi-edid-dumps

```diff
@@ -70,107 +101,363 @@ def run_cmd(cmd):
+            return RC_ERR_SUBPROCESS
+
+        flashed_screen = True
+
+        # Generate the RPi and kano-settings specific data.
+        edid_dumps = generate_rpi_dumps(raw_edid_path, edid_dumps_path)
+        if not edid_dumps:
+            return RC_ERR_SUBPROCESS
+
+        if not generate_expected_data(edid_dumps, expected_data_path):
+            return RC_ERR_SUBPROCESS
+
+    if flashed_screen:
+        if not restore_original_edid():
+            return RC_ERR_SUBPROCESS
+
+    return RC_SUCCESS
+
+
+def check_python_modules(modules):
+    """ Check that the given Python modules are installed. """
+
+    has_all_modules = True
+
+    for module in modules:
+        try:
+            importlib.import_module(module)
+        except ImportError:
+            sys.stderr.write('Error: Missing dependency on Python module {}\n'.format(module))
+            has_all_modules = False
+
+    return has_all_modules
+
+
+def check_installed_tools(binary_paths):
+    """ Check that the given binaries are installed. """
+
+    has_all_binaries = True
+
+    for binary_path in binary_paths:
+        if not os.path.exists(binary_path):
+            sys.stderr.write('Error: Missing dependency on binary {}\n'.format(binary_path))
+            has_all_binaries = False
+
+    return has_all_binaries
+
+
+def check_boot_config_txt():
+    """ Check that the /boot/config.txt is setup correctly for this tool to work. """
+
+    valid_config_txt = True
+
+    with open('/boot/config.txt', 'r') as config_txt:
+        config_txt_lines = config_txt.readlines()
+
+    if 'hdmi_force_hotplug=1\n' in config_txt_lines:
+        sys.stderr.write(
+            'Error: hdmi_force_hotplug=1 was found in /boot/config.txt\n'
+        )
+        valid_config_txt = False
+
+    if 'dtparam=i2c2_iknowwhatimdoing=on\n' not in config_txt_lines:
+        sys.stderr.write(
+            'Error: dtparam=i2c2_iknowwhatimdoing=on was not found in /boot/config.txt\n'
+        )
+        valid_config_txt = False
+
+    return valid_config_txt
+
+
+def check_i2c_bus(bus_number):
+    """ Check that the i2c device at address 3f805000.i2c is available. """
+
+    i2c_bus_available = True
+
+    import smbus
+    try:
+        smbus.SMBus(bus_number)
+    except:
+        sys.stderr.write(
+            'Error: Could not find the i2c device at address 3f805000.i2c, bus #2.'
+            ' Run sudo i2cdetect -l for more info.\n'
+        )
+        i2c_bus_available = False
+
+    return i2c_bus_available
+
+
+def check_dependencies():
+    """ Check all dependencies for this tool to run on a RPi. """
+
+    return (
+        check_python_modules(REQUIRED_PYTHON_MODULES) and
+        check_installed_tools(REQUIRED_TOOLS) and
+        check_boot_config_txt() and
+        check_i2c_bus(RPI_HDMI_I2C_BUS_NUM)
+    )
+
 
 if __name__ == '__main__':
-    if os.geteuid() == 0:
-        print 'Program must NOT be run as root.'
-        sys.exit(RC_NOT_USER)
+    if os.geteuid() != 0:
```

Comment:
I am not sure that we need to `set-uid` this tool.

Reply:
Yes, did that one again... Good spot! Fixed